apps/dc_tools/odc/apps/dc_tools/stac_api_to_dc.py

```diff
@@ -45,7 +45,9 @@ def _parse_options(options: Optional[str]) -> Dict[str, Any]:
     return parsed_options
 
 
-def _guess_location(item: pystac.Item) -> Tuple[str, bool]:
+def _guess_location(
+    item: pystac.Item, rewrite: Optional[Tuple[str, str]] = None
```

Comment:
Just replacing prefix of a URL with another is not always sufficient, I would recommend rewrite to have type
`: Optional[Callable[[str], str]]`. For `http[s]->s3` one needs to undo url encoding applied to the path component and not just replace `https://some.host.tld/maybe_prefix/` with `s3://bucket/`. For example character `+` is encoded as `%2B` in http urls, but is just a `+` in S3 urls.

Reply:
How would you pass that in from the command line?

Reply:
you won't, the only supported method would be `http->s3` (and maybe reverse?), and configuration for that would be "prefix" to replace and with "what", but it will also need to call `from urllib.parse import unquote` on the path. Although I'm not sure what characters are allowed in s3 urls, I know for sure that issue `+` vs `%2B` exists.

Reply:
in http url `+` is ` `, but in `s3` url `+` is just `+`, and ` ` is allowed as is too.